src/App.jsx

```diff
@@ -19,11 +19,14 @@ export function App() {
 	 * to create and join a new list.
 	 */
 	const [listToken, setListToken] = useStateWithStorage(
-		'my test list',
+		null,
 		'tcl-shopping-list-token',
 	);
 
 	useEffect(() => {
+		if (listToken) {
```

Comment:
This is creating an infinite loop. Creating a new token can be handled on the Home view (see comments there).

Reply:
I've looked at this again and I think you're right about the token check happening on the Home view rather than the App component, it's definitely redundant so I will be removing the redirect command there. However, I'm not sure if it's causing infinite loops as it's listening for changes in the listToken state rather than the entire page/app.

Reply:
I think keeping the navigate in the Home view as you said will work. 
From what I understand: useEffect from the App component is constantly redirecting to the list page. However, if it's on the homepage, it only redirects once. 
I think this will solve the issue.

Reply:
Because I specified a dependency array, the useEffect will not run after the first re-render unless the value in the array changes. I've removed redirect as it does not have to be in the App component, but I'm confident that it wasn't causing infinite redirects to the list page as there were no constant changes to listToken. As of right now I've yet to find out what is causing the spike in the database reads but I can check the Network tab to see what requests are getting sent out, I think that can be a good start.